kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/builders.kt

```diff
@@ -56,12 +90,36 @@ fun <A> arbitrary(
       override fun sample(rs: RandomSource): Sample<A> = sampleOf(sampleFn(rs), shrinker)
    }
 
-/**
- * Creates a new [Arb] that performs shrinking using the supplied [Shrinker], has no edge cases and
- * generates values from the given function.
- */
-fun <A> arbitrary(shrinker: Shrinker<A>, fn: (RandomSource) -> A): Arb<A> =
-   arbitrary(emptyList(), shrinker, fn)
+fun <A> arbitraryBuilder(
+   shrinker: Shrinker<A>? = null,
+   classifier: Classifier<A>? = null,
+   edgecaseFn: EdgecaseFn<A>? = null,
+   builderFn: suspend ArbitraryBuilderSyntax.(RandomSource) -> A
+): Arb<A> = object : Arb<A>() {
+   override fun edgecase(rs: RandomSource): A? = computeArb().edgecase(rs)
+   override fun sample(rs: RandomSource): Sample<A> = computeArb().sample(rs)
+
+   private fun computeArb(): Arb<A> {
+      val continuation = ArbContinuation<A>()
+      val wrapReturn: suspend ArbitraryBuilderSyntax.() -> Arb<A> = {
+         val value: A = builderFn(randomSource.bind())
+         ArbitraryBuilder(
+            { value },
+            classifier,
+            shrinker,
+            { rs ->
+               // use edgecase function if provided, otherwise retain those from flatmap compositions
+               if (edgecaseFn != null) edgecaseFn.invoke(rs) else value
```

Comment:
this should always default to value

Reply:
done! added test in https://github.com/kotest/kotest/pull/2494/files#diff-2e6b0f63218b6d89d463623c97b883bb2d959b400807214765432ebf17f615ffR88-R107